android/src/main/java/bbct/android/common/activity/MainActivity.java

```diff
@@ -165,19 +167,44 @@ public boolean onCreateOptionsMenu(Menu menu) {
         return super.onCreateOptionsMenu(menu);
     }
 
+    @Override
+    public boolean onSupportNavigateUp() {
+        this.onBackPressed();
+        return true;
+    }
+
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        int itemId = item.getItemId();
+        int menuId = item.getItemId();
+        switch (menuId) {
+            case R.id.about_menu:
+                this.getSupportFragmentManager()
+                        .beginTransaction()
+                        .replace(R.id.fragment_holder, new About(), FragmentTags.ABOUT)
+                        .addToBackStack(FragmentTags.ABOUT)
+                        .commit();
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
 
-        if (itemId == R.id.about_menu) {
-            this.getSupportFragmentManager()
-                    .beginTransaction()
-                    .replace(R.id.fragment_holder, new About(), FragmentTags.ABOUT)
-                    .addToBackStack(FragmentTags.ABOUT)
-                    .commit();
-            return true;
+    @Override
+    public void onBackStackChanged() {
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(getSupportFragmentManager().getBackStackEntryCount() > 1);
```

Comment:
It can only be false if we change the order of the fragment stack, for example instead of adding the BaseballCardList first, we would add the BaseballCardDetail, same on previous implementation.